servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/RequestResponseCloseHandler.java

```diff
@@ -163,6 +163,13 @@ public void protocolPayloadBeginOutbound(final ChannelHandlerContext ctx) {
 
     @Override
     public void protocolPayloadEndOutbound(final ChannelHandlerContext ctx) {
+        if (isClient || has(state, CLOSING)) {
```

Comment:
Unless I misunderstood our prior discussing, I assumed you suggested implementing it in `RequestResponseCloseHandler` as opposed to having a more elaborate state machine inside `WriteStreamSubscriber`.  What was your expectation?

Reply:
I am not questioning the changes to `RequestResponseCloseHandler` but why do we need the special case for `isClient`. Is `has(state, CLOSING)` sufficient?

Reply:
I feel like there shouldn't be any condition at all, not `isClient()` nor `has(CLOSING)`. `end of payload` should mean what it is and emit every time the codec sees one. 
But if we did that, then we'd get into the issue down the line with H1 repeated read/single write reusing a terminated `WriteStreamSubscriber`. So we have to suppress this event before it reaches `WSS` in the case of H1 server; for H2 it doesn't matter.

Reply:
I understand why `has(CLOSING)` is required and I think it is reasonable as this event is only valuable to determine if we expect any other write on this connection. Since `CloseHandler` has all the other information (outbound closing) it makes sense to keep this knowledge inside `CloseHandler`.

I am not clear about why we need `isClient` check here, can you explain?

Reply:
The discrepancy is the client `WriteStreamSubscriber` lifetime is tied to the request where on the server side its lifetime is aligned with the connection. Hence to be correct the client needs to trigger this event for every write, no matter if it'll be the last write on the connection. 
In practice it doesn't matter because close handler does proper sequencing and attaches listeners to the write promise so we can't ever see a premature channel closed event.

Reply:
Also `has(CLOSING)` is tied to the connection, not the lifetime of `WSS` which differs between client and server.

Reply:
Ok ya makese sense, thanks for the explanation.